storage/innobase/buf/buf0buf.cc

```diff
@@ -847,8 +847,17 @@ class mem_pressure
   {
     uint64_t u= 1;
     m_abort= true;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
+#pragma GCC diagnostic ignored "-Wempty-body"
+    /* return result ignored, cannot do anything with it */
     if (write(m_event_fd, &u, sizeof(uint64_t)) != sizeof(uint64_t))
```

Comment:
If we remove the `if` as well, we can remove the `-Wempty-body` and simplify this.